src/devices/cpu/drcbearm64.h

```diff
@@ -0,0 +1,286 @@
+// license:BSD-3-Clause
+// copyright-holders:windyfairy
+#ifndef MAME_CPU_DRCBEARM64_H
+#define MAME_CPU_DRCBEARM64_H
+
+#pragma once
+
+#include "drcuml.h"
+#include "drcbeut.h"
+
+#include "asmjit/src/asmjit/asmjit.h"
+#include "asmjit/src/asmjit/a64.h"
+
+#include <vector>
+
+
+namespace drc {
+
+class drcbe_arm64 : public drcbe_interface
+{
+	typedef uint32_t (*arm64_entry_point_func)(void *entry);
+
+public:
+	drcbe_arm64(drcuml_state &drcuml, device_t &device, drc_cache &cache, uint32_t flags, int modes, int addrbits, int ignorebits);
+	virtual ~drcbe_arm64();
+
+	virtual void reset() override;
+	virtual int execute(uml::code_handle &entry) override;
+	virtual void generate(drcuml_block &block, const uml::instruction *instlist, uint32_t numinst) override;
+	virtual bool hash_exists(uint32_t mode, uint32_t pc) override;
+	virtual void get_info(drcbe_info &info) override;
+	virtual bool logging() const override { return false; }
+
+private:
+	class be_parameter
+	{
+		static int const REG_MAX = 30;
+
+	public:
+		// parameter types
+		enum be_parameter_type
+		{
+			PTYPE_NONE = 0,        // invalid
+			PTYPE_IMMEDIATE,       // immediate; value = sign-extended to 64 bits
+			PTYPE_INT_REGISTER,    // integer register; value = 0-REG_MAX
+			PTYPE_FLOAT_REGISTER,  // floating point register; value = 0-REG_MAX
+			PTYPE_MEMORY,          // memory; value = pointer to memory
+			PTYPE_MAX
+		};
+
+		typedef uint64_t be_parameter_value;
+
+		be_parameter() : m_type(PTYPE_NONE), m_value(0) { }
+		be_parameter(const be_parameter &param) : m_type(param.m_type), m_value(param.m_value) { }
+		be_parameter(uint64_t val) : m_type(PTYPE_IMMEDIATE), m_value(val) { }
+		be_parameter(drcbe_arm64 &drcbe, const uml::parameter &param, uint32_t allowed);
```

Comment:
Do you actually need the explicit copy constructor here?  It’s just doing a memberwise copy, isn’t it?  Can you just do:
```c++
		be_parameter(const be_parameter &param) = default;
```